js/ui/status/automaticUpdates.js

```diff
@@ -0,0 +1,301 @@
+        let icon = this._getIcon()
+
+        this._item.icon.gicon = icon;
+        this._indicator.gicon = icon;
+
+        // Only show the Automatic Updates icon at the bottom bar when it is
+        // both enabled, and there are updates being downloaded or installed.
+        this._updateVisibility();
+
+        // The status label
+        this._item.label.text = _("Automatic Updates");
+    },
+
+    _updateActiveConnection: function() {
+        let currentActiveConnection = this._getActiveConnection();
+
+        if (this._activeConnection == currentActiveConnection)
+            return;
+
+        // Disconnect from the previous active connection
+        if (this._settingChangedSignalId)
+            this._activeConnection.disconnect(this._userSettingSignalId);
+
+        this._activeConnection = currentActiveConnection;
+
+        // Connect from the current active connection
+        if (currentActiveConnection)
+            this._settingChangedSignalId = currentActiveConnection.connect('changed', Lang.bind(this, this._sync));
+    },
+
+    _updateAutomaticUpdatesItem: function() {
+        let state = this._getState();
+
+        if (state == AutomaticUpdatesState.DISABLED)
+            this._toggleItem.label.text = _("Turn Automatic Updates ON");
+        else
+            this._toggleItem.label.text = _("Turn Automatic Updates OFF");
+    },
+
+    _toggleAutomaticUpdates: function() {
+        if (!this._activeConnection)
+            return;
+
+        let userSetting = this._ensureUserSetting(this._activeConnection);
+
+        // The string representation here is the oposite of the value
+        let value = '0';
+        if (userSetting.get_data(NM_SETTING_ALLOW_DOWNLOADS_WHEN_METERED) != '1')
+            value = '1';
+
+        userSetting.set_data(NM_SETTING_ALLOW_DOWNLOADS_WHEN_METERED, value);
+
+        this._activeConnection.commit_changes_async(true, null, (con, res, data) => {
+            this._activeConnection.commit_changes_finish(res);
+            this._sync();
+        });
+    },
+
+    _ensureUserSetting: function(connection) {
+        let userSetting = connection.get_setting(NM.SettingUser.$gtype);
+        if (!userSetting) {
+            userSetting = new NM.SettingUser();
+            activeConnection.add_setting(userSetting);
+            connection.commit_changes(true, null);
+        }
+        return userSetting;
+    },
+
+    _getIcon: function() {
+        let state = this._getState();
+        let iconName = automaticUpdatesStateToString(state);
+
+        if (!iconName)
+            return null;
+
+        let iconFile = Gio.File.new_for_uri(iconName);
+        let gicon = new Gio.FileIcon({ file: iconFile });
+
+        return gicon;
+    },
+
+    _updateVisibility: function() {
+        let state = this._getState();
+
+        this._item.actor.visible = state != AutomaticUpdatesState.DISCONNECTED;
+        this._indicator.visible = state == AutomaticUpdatesState.DOWNLOADING;
```

Comment:
I know it's not necessary, but I'd add parenthesis here for the sake of readability)